test/data/upgrade/upgrade_prepare.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# This upgrades the system to the latest RHEL-Y
+set -xeuo pipefail
+
+source /etc/os-release
+source define-compose-url.sh
+
+# Disable gpgcheck for internal repositories
+echo "gpgcheck=0" >> /etc/yum.repos.d/baseos.repo
+echo "gpgcheck=0" >> /etc/yum.repos.d/appstream.repo
+
+# Install SUT
+dnf install -y osbuild-composer composer-cli
+
+# Expecting either 8 or 9
+if [[ "$VERSION_ID" == "8" ]]; then
+  # Prepare the upgrade
+  curl -k -o /etc/yum.repos.d/oam-group-leapp-rhel-8.repo https://gitlab.cee.redhat.com/oamg/upgrades-dev/oamg-rhel8-vagrant/-/raw/main/roles/init/files/leapp-copr.repo
+  # install the leapp upgrade tool + other dependencies
+  dnf install -y leapp-upgrade-el8toel9 vdo jq rpmdevtools
+else
```

Comment:
Maybe also test for 9 and error if it isn't either to future-proof this against being run on 10 before it's ready?